src/parser/binary-operation-parser.ts

```diff
@@ -64,108 +57,100 @@ export type HavingGrouper<DB, TB extends keyof DB> = (
   qb: HavingInterface<DB, TB>
 ) => HavingInterface<DB, TB>
 
-export type FilterOperator = Operator | Expression<any>
+export type ComparisonOperatorExpression = ComparisonOperator | Expression<any>
 
-type FilterType = 'where' | 'having' | 'on'
+type FilterExpressionType = 'where' | 'having' | 'on'
 
-export function parseWhereFilter(args: any[]): OperationNode {
-  return parseFilter('where', args)
+export function parseWhere(args: any[]): OperationNode {
+  return parseFilterExpression('where', args)
 }
 
-export function parseHavingFilter(args: any[]): OperationNode {
-  return parseFilter('having', args)
+export function parseHaving(args: any[]): OperationNode {
+  return parseFilterExpression('having', args)
 }
 
-export function parseOnFilter(args: any[]): OperationNode {
-  return parseFilter('on', args)
+export function parseOn(args: any[]): OperationNode {
+  return parseFilterExpression('on', args)
 }
 
-export function parseReferenceFilter(
-  lhs: ReferenceExpression<any, any>,
-  op: FilterOperator,
-  rhs: ReferenceExpression<any, any>
-): FilterNode {
-  return FilterNode.create(
-    parseReferenceExpression(lhs),
-    parseFilterOperator(op),
-    parseReferenceExpression(rhs)
+export function parseReferentialBinaryOperation(
```

Comment:
This is also now `parseReferentialFilter` since it uses `parseComparisonOperatorExpression`.

Reply:
Was also thinking `parseReferentialComparisonOperation`, but will go with `parseReferentialFilter`.